rita_exit/src/database/mod.rs

```diff
@@ -517,3 +526,88 @@ pub fn enforce_exit_clients(client_data: &mut RitaExitData) -> Result<(), Box<Ri
     client_data.set_debt_actions(new_debt_actions);
     Ok(())
 }
+
+/// Removes client ip assignments for clients that have not checked in for 1 hour
+pub fn teardown_inactive_clients(client_data: &mut RitaExitData) {
+    let inactive_list = client_data.get_inactive_list();
+    // check for snat mode
+    let ipv4_mode = client_data.get_ipv4_nat_mode();
+    if let ExitIpv4RoutingSettings::SNAT { .. } = ipv4_mode {
+    } else {
+        return;
+    }
+    let ext_nic = &settings::get_rita_exit().network.external_nic.unwrap();
+    let ext_assignments = client_data.get_external_ip_assignments();
+    let int_assignments = client_data.get_internal_ip_assignments();
+    // for each client in inactive, find their wg key from clientandipmap, then
+    // remove their iptables rules if their last seen was over threshold
+    for (client, last_seen) in inactive_list.iter() {
+        if last_seen.elapsed().as_secs() > INACTIVE_CLIENT_THRESHOLD {
+            try_teardown_client_snat(
+                ext_assignments.clone(),
+                int_assignments.clone(),
+                *client,
+                ext_nic,
```

Comment:
why not merge the lists and have one teardown loop, I'm also not seeing how exactly the client's ip's get removed from the active list so that once torn down they can be used again but maybe I'm not reading it right.

Reply:
you're right, this was never being handled. added in removal from the external_ip_assignments list to the teardown function.